src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultImmutableConverter.cs

```diff
@@ -47,21 +55,44 @@ private string GetConstructingTypeName(string immutableCollectionTypeName)
                     return ImmutableListTypeName;
                 case ImmutableStackGenericTypeName:
                 case ImmutableStackGenericInterfaceTypeName:
-                    return  ImmutableStackTypeName;
+                    return ImmutableStackTypeName;
                 case ImmutableQueueGenericTypeName:
                 case ImmutableQueueGenericInterfaceTypeName:
-                    return  ImmutableQueueTypeName;
+                    return ImmutableQueueTypeName;
                 case ImmutableSortedSetGenericTypeName:
-                    return  ImmutableSortedSetTypeName;
+                    return ImmutableSortedSetTypeName;
                 case ImmutableHashSetGenericTypeName:
                 case ImmutableSetGenericInterfaceTypeName:
-                    return  ImmutableHashSetTypeName;
+                    return ImmutableHashSetTypeName;
+                case ImmutableDictionaryGenericTypeName:
+                case ImmutableDictionaryGenericInterfaceTypeName:
+                    return ImmutableDictionaryTypeName;
+                case ImmutableSortedDictionaryGenericTypeName:
+                    return ImmutableSortedDictionaryTypeName;
                 default:
                     // TODO: Refactor exception throw following serialization exception changes. 
                     throw new NotSupportedException(SR.Format(SR.DeserializeTypeNotSupported, immutableCollectionTypeName));
```

Comment:
@steveharter - is this actually a TODO? NotSupportedException seems right.

Reply:
Agreed. The plan was to utilize the newly added  [`NotSupportedException GetNotSupportedException_SerializationNotSupportedCollection`](https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs#L21-L29). I'll update this.